### PR TITLE
NO-JIRA: Update tuned profile degraded test

### PR DESCRIPTION
**Pull request by rbaturov**
Updated the tuned degradation test to fail only when the tuned profile was found degraded
on BM host.
In case of tuned profile found degraded on a VM, only a warning would be reported in the logs.
The reason for that is that the fact CI is using VMs - and some
configurations like certain kernel args can't be applied and therby lead to
the tuned profile being degraded.

### CODE REVIEW

**Comment by jmencak** [comment]
/approve
/lgtm

Thank you for the PR.

**Comment by yanirq** [comment]
/retest
check flakiness.

**Comment by rbaturov** [comment]
@jmencak @yanirq 
Can you please review the newly applied changes?

File: test/e2e/performanceprofile/functests/utils/infrastructure/vm.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+// IsVM checks if a given node's underlying infrastructure is a VM
+func IsVM(node *corev1.Node) (bool, error) {
+	cmd := []string{
+		"/bin/bash",
+		"-c",
+		"systemd-detect-virt > /dev/null ; echo $?",
+	}
+	output, err := nodes.ExecCommandOnMachineConfigDaemon(context.TODO(), node, cmd)
```
[the remark this answers is not in the document]
Comment (rbaturov) [reply test/e2e/performanceprofile/functests/utils/infrastructure/vm.go:19]: Currently, there is no alternative, that's how we are executing commands to retrieve host info. (correct me if I'm wrong).
I'm actually working on a PR at these moments,  that will launch daemonset with a privileged pod and hostsfs mounted in order to replace that (MCD), that would be deployed with test suites. For this I would need to refactor all `ExecCommandOnMachineConfigDaemon` calls, including this one.

[the remark this answers is not in the document]
Comment (Tal-or) [reply test/e2e/performanceprofile/functests/utils/infrastructure/vm.go:19]: @yanirq We should wait for Ronny's PR

[the remark this answers is not in the document]
Comment (Tal-or) [reply test/e2e/performanceprofile/functests/utils/infrastructure/vm.go:19]: No rush BTW, there are plenty of other blockers

**Comment by yanirq** [comment]
/retitle NO-JIRA: Update tuned profile degraded test

File: test/e2e/performanceprofile/functests/1_performance/performance.go [hunk 2]
```diff
@@ -126,7 +127,13 @@ var _ = Describe("[rfe_id:27368][performance]", Ordered, func() {
 				Expect(err).ToNot(HaveOccurred(), "Failed to get the Tuned profile for node %s", node.Name)
 				degradedCondition := findCondition(tunedProfile.Status.Conditions, "Degraded")
 				Expect(degradedCondition).ToNot(BeNil(), "Degraded condition not found in Tuned profile status")
-				Expect(degradedCondition.Status).To(Equal(corev1.ConditionFalse), "Tuned profile is degraded")
+				isNodeBasedOnVM, err := infrastructure.IsVM(&node)
+				Expect(err).ToNot(HaveOccurred(), "Failed to detect if the node is based on VM")
+				if isNodeBasedOnVM {
+					testlog.Warning(fmt.Sprintf("Tuned profile is degraded. A warning raised as the node is based on a VM, and is expected. Error message: %s", degradedCondition.Message))
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/1_performance/performance.go:133]: not all failures here are  expected,  please remove "and is expected"

Comment (rbaturov) [reply test/e2e/performanceprofile/functests/1_performance/performance.go]: Done

**Pull request merged**